### ISSUE TITLE
模型差役问题，求指导。

### ISSUE THREAD

**Issue by lovetian1991** [opendatalab/DocLayout-YOLO#167]
<img width="545" height="212" alt="Image" src="https://github.com/user-attachments/assets/bb2eadcb-10a4-4db5-972b-0dfaa161358b" />
mineru使用的模型只有30几M，训练的模型100多m，但是推理测试doclayout_yolo_docsynth300k_imgsz1600这个模型效果不行，我如果需要微调的话，直接基于mineru作为预训练模型微调可以么。大佬

**Comment by JulioZhao97** [comment]
@lovetian1991 可以的，但是你要用pypi上最新的代码，这里的没更新

**Comment by lovetian1991** [comment]
<img width="1014" height="749" alt="Image" src="https://github.com/user-attachments/assets/1f083cfb-54a0-4bea-92ed-3243bd36d057" />大佬帮忙看看，用源码训练的时候会说权重兼容性有问题，这个是代码问题么？

**Comment by lovetian1991** [comment]
@JulioZhao97 大佬帮忙在看下 具体是pypi哪个代码，是https://pypi.org/project/doclayout-yolo/#files这个么

**Comment by lovetian1991** [comment]
@JulioZhao97 更新0.0.4的代码后 还是这个错误，求指导

**Comment by JulioZhao97** [comment]
@lovetian1991 您好，这个问题只有在做最后的final evaluation的时候会报出来，正常训练的中间epoch都是可以的，不影响，可以看一下您的训练script吗

**Comment by lovetian1991** [comment]
@JulioZhao97  用如下代码可以解决，不知道有啥影响没？

def dilated_conv(self, x, dilation):
act = self.dcv.act
weight = self.dcv.conv.weight
padding = dilation * (self.k // 2)

// 检查 bn 是否存在
if hasattr(self.dcv, 'bn') and self.dcv.bn is not None:
    return act(self.dcv.bn(F.conv2d(x, weight, stride=1, padding=padding, dilation=dilation)))
else:
    //print('without_bn', self.dcv)
    return act(F.conv2d(x, weight, stride=1, padding=padding, dilation=dilation))

**Comment by lovetian1991** [comment]
训练脚本python3 train.py --epoch=500 --lr0=0.02 --model=ts --batch-size=4 --project=runs/train --image-size=1280 --data=/app/data --pretrain=/app/model/doclayout_yolo_docstructbench_imgsz1280_2501.pt --optimizer=SGD --device=1,2

**Comment by JulioZhao97** [comment]
> [@JulioZhao97](https://github.com/JulioZhao97) 用如下代码可以解决，不知道有啥影响没？
> 
> def dilated_conv(self, x, dilation): act = self.dcv.act weight = self.dcv.conv.weight padding = dilation * (self.k // 2)
> 
> // 检查 bn 是否存在 if hasattr(self.dcv, 'bn') and self.dcv.bn is not None: return act(self.dcv.bn(F.conv2d(x, weight, stride=1, padding=padding, dilation=dilation))) else: //print('without_bn', self.dcv) return act(F.conv2d(x, weight, stride=1, padding=padding, dilation=dilation))

应该是没什么影响，您报这个错应该是最后validation的时候报错吧，训练中是不会报这个错的

**Comment by lovetian1991** [comment]
@JulioZhao97 执行训练命令后，训练几轮后就会报这个错误，最终没有输出最终的权重文件。

**Comment by lovetian1991** [comment]
@JulioZhao97 调整这个代码后，可以训练出来了，非常感谢大佬。求给这个issue给点思路[169](https://github.com/opendatalab/DocLayout-YOLO/issues/169)
